pkg/types/validation/installconfig.go

```diff
@@ -462,22 +463,25 @@ func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Required(fldPath, "must include httpProxy or httpsProxy"))
 	}
 	if p.HTTPProxy != "" {
-		if err := validate.URI(p.HTTPProxy); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpProxy"), p.HTTPProxy, err.Error()))
+		if err := validateURI(p.HTTPProxy, fldPath.Child("httpProxy"), []string{"http"}); err != nil {
+			allErrs = append(allErrs, err...)
 		}
 	}
 	if p.HTTPSProxy != "" {
-		if err := validate.URI(p.HTTPSProxy); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("httpsProxy"), p.HTTPSProxy, err.Error()))
+		if err := validateURI(p.HTTPSProxy, fldPath.Child("httpsProxy"), []string{"http", "https"}); err != nil {
+			allErrs = append(allErrs, err...)
 		}
 	}
 	if p.NoProxy != "" && p.NoProxy != "*" {
-		for idx, v := range strings.Split(p.NoProxy, ",") {
-			v = strings.TrimSpace(v)
-			errDomain := validate.NoProxyDomainName(v)
-			_, _, errCIDR := net.ParseCIDR(v)
-			if errDomain != nil && errCIDR != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("noProxy").Index(idx), v, "must be a CIDR or domain, without wildcard characters"))
+		if strings.Contains(p.NoProxy, " ") {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("noProxy"), p.NoProxy, fmt.Sprintf("noProxy must not have spaces")))
+		} else {
+			for idx, v := range strings.Split(p.NoProxy, ",") {
+				errDomain := validate.NoProxyDomainName(v)
+				_, _, errCIDR := net.ParseCIDR(v)
+				if errDomain != nil && errCIDR != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("noProxy").Index(idx), v, "must be a CIDR or domain, without wildcard characters"))
+				}
```

Comment:
I would consider continuing the validation of the individual elements even if there are spaces.
```suggestion
		for idx, v := range strings.Split(p.NoProxy, ",") {
			if trimmedV := strings.Trim(v); v != trimmedV {
				allErrs = append(allErrs, field.Invalid(fldPath.Child("noProxy"), p.NoProxy, fmt.Sprintf("noProxy must not have spaces, which are present in element %d %q", idx, v)))
				v = trimmedV
			}
			errDomain := validate.NoProxyDomainName(v)
			_, _, errCIDR := net.ParseCIDR(v)
			if errDomain != nil && errCIDR != nil {
				allErrs = append(allErrs, field.Invalid(fldPath.Child("noProxy").Index(idx), v, "must be a CIDR or domain, without wildcard characters"))
		}
```

Reply:
I feel checking for spaces once outside the for loop would be better.